portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/FavoritesHandler.java

```diff
@@ -124,58 +129,121 @@ public int doPost(String[] parts, HttpServletRequest req,
 		return NEXT;
 	}
 
-	public UserFavorites userFavorites(String userId)
-		throws PermissionException, PortalHandlerException, InUseException, IdUnusedException {
-		UserFavorites result = new UserFavorites();
+	public FavoriteSites getUserFavorites(String userId) {
+		FavoriteSites favoriteSites = new FavoriteSites();
+		favoriteSites.favoriteSiteIds = portalService.getPinnedSites(userId);
+		return favoriteSites;
+	}
 
-		if (userId == null) {
-			// User isn't logged in
-			return result;
-		}
+	public void updateUserFavorites(final String userId) {
+		if (StringUtils.isBlank(userId)) return;
 
-		List<String> existingHiddenSiteIds = Collections.<String>emptyList();
 		Preferences prefs = preferencesService.getPreferences(userId);
 		ResourceProperties props = prefs.getProperties(PreferencesService.SITENAV_PREFS_KEY);
-		List propList = props.getPropertyList(PreferencesService.SITENAV_PREFS_EXCLUDE_KEY);
-		if (propList != null) {
-			existingHiddenSiteIds = (List<String>) propList;
-		}
+		List<String> excludedSites = Optional.ofNullable(props.getPropertyList(PreferencesService.SITENAV_PREFS_EXCLUDE_KEY)).orElseGet(Collections::emptyList);
+
+		List<String> pinnedSites = portalService.getPinnedSites(userId);
+		List<String> unPinnedSites =  portalService.getUnpinnedSites(userId);
+		List<String> combinedSiteIds = Stream.concat(pinnedSites.stream(), unPinnedSites.stream()).collect(Collectors.toList());
+
+		// when the user has no sites it is most likely their first login since pinning was introduced
+		if (combinedSiteIds.isEmpty()) {
+			log.debug("User has no pinned site data performing favorites migration for user [{}]", userId);
+			// look to the users favorites stored in preferences
+			List<String> favoriteSiteIds = props.getPropertyList(FAVORITES_PROPERTY);
+			if (favoriteSiteIds != null && !favoriteSiteIds.isEmpty()) {
```

Comment:
Use StringUtils.isBlank or StringUtils.isBlank

Reply:
I think StringUtils is only for Strings, no?

Reply:
Oops... Yesterday I happened to see that it was String.

Reply:
Maybe CollectionUtils.isNotEmpty?